api_tests/src/user.spec.ts

```diff
@@ -101,20 +104,37 @@ test("Delete user", async () => {
     .comment;
   expect(remoteComment).toBeDefined();
 
+  // Fetch user content before deletion
+  let betaPost1 = (await resolvePost(beta, localPost)).post;
+  expect(betaPost1).toBeDefined();
+  let betaPost2 = (await resolvePost(beta, remotePost)).post;
+  expect(betaPost2).toBeDefined();
+
+  // Delete user account and content
   await deleteUser(user);
 
-  await expect(resolvePost(alpha, localPost)).rejects.toBe(
+  // Attempt to fetch user content from original instance, fails because its deleted
+  await expect(resolvePost(gamma, localPost)).rejects.toBe(
     "couldnt_find_object",
   );
-  await expect(resolveComment(alpha, localComment)).rejects.toBe(
+  await expect(resolveComment(gamma, localComment)).rejects.toBe(
     "couldnt_find_object",
   );
-  await expect(resolvePost(alpha, remotePost)).rejects.toBe(
+  await expect(resolvePost(gamma, remotePost)).rejects.toBe(
     "couldnt_find_object",
   );
-  await expect(resolveComment(alpha, remoteComment)).rejects.toBe(
+  await expect(resolveComment(gamma, remoteComment)).rejects.toBe(
     "couldnt_find_object",
   );
+
+  // Attempt to read user content from federated instance, fails because of federated
+  // deletion
+  expect((await getPost(beta, betaPost1!.post.id)).post_view.post.deleted).toBe(
+    true,
+  );
+  expect((await getPost(beta, betaPost2!.post.id)).post_view.post.deleted).toBe(
+    true,
+  );
```

Comment:
This test was incorrect before as it always refetched the objects from alpha which would return an error. But it didnt check that the delete was properly sent from alpha to beta to delete objects which were fetched previously.